packages/flutter/lib/src/services/asset_resolver.dart

```diff
@@ -0,0 +1,70 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:mojo/core.dart' as core;
+
+import 'asset_bundle.dart';
+
+/// Base class for asset resolvers.
+abstract class AssetResolver {
+  /// Return a resolved asset key for the asset named [name].
+  Future<String> resolve(String name);
+}
+
+/// Builder for a particular AssetResolver.
+///
+/// Allows AssetResolvers to be specified in widget build() functions.
+/// Subclasses should override operator== and hashCode so that equivalent
+/// configurations appear to be equal. Builder objects should also be
+/// cheap to construct.
+abstract class AssetResolverBuilder<T extends AssetResolver> {
+  const AssetResolverBuilder();
+  T buildResolver();
+}
```

Comment:
Can you point to an example of "other delegates of this type" that you mention above?